### PR TITLE
Epamlab webapp lowcoster finish dao

### PR DESCRIPTION
**Pull request by klysov**
DAO finished

### CODE REVIEW

File: src/main/java/dao/CustomerDAO.java [hunk 1]
```diff
@@ -65,14 +65,11 @@ public boolean insert(Customer customer) throws SQLException {
...
         Connection connection = ConnectionFactory.getConnection();
-        PreparedStatement preparedStatement = connection.prepareStatement("UPDATE customer " +
-                "SET Name=?, LastName=?, PersonID=?" +
-                " WHERE idCustomer=?");
-
-        preparedStatement.setInt(1, customer.getCustomerId());
-        preparedStatement.setString(2, customer.getCustomerFirstName());
-        preparedStatement.setString(3, customer.getCustomerLastName());
-        preparedStatement.setString(4, customer.getCustomerPersonId());
+        PreparedStatement preparedStatement = connection.prepareStatement("UPDATE Customer SET Name=?, LastName=?, PersonID=? WHERE idCustomer=?");
```
Comment (SMishAlex) [inline src/main/java/dao/CustomerDAO.java]: please change column name to first name too

File: src/main/java/dao/FlightDAO.java [hunk 2]
```diff
@@ -1,33 +1,88 @@
...
+import java.util.HashSet;
 import java.util.Set;
 
 public class FlightDAO implements DAO<Flight> {
+
+    private Flight extractFlightFromResultSet(ResultSet resultSet) throws SQLException {
+        Flight flight = new Flight();
+        flight.setFlightId( resultSet.getInt("idFlight") );
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+        flight.setDateTime(LocalDateTime.from(dateTimeFormatter.parse(resultSet.getString("DateTime").replaceAll("\\.0",""))));
```
Comment (SMishAlex) [inline src/main/java/dao/FlightDAO.java]: too many expressions in one line... it's better to make some variables to increase readability

File: src/test/java/dao/CustomerDAOTest.java [hunk 3]
```diff
@@ -1,4 +1,93 @@
...
+    public static void setupDatabase() throws SQLException {
+        Connection connection = ConnectionFactory.getConnection();
+        for (int i = 0; i < NUMBER_OF_ENTRIES; i++) {
+            PreparedStatement preparedStatement =
+                    connection.prepareStatement("INSERT INTO Customer (Name, LastName, PersonID) VALUES(?, ?, ?)");
+            preparedStatement.setString(1, "NAME_ENTRY#" + i);
+            preparedStatement.setString(2, "LASTNAME_ENTRY#" + i);
+            preparedStatement.setString(3, "PERSONID_ENTRY#" + i);
+            preparedStatement.executeUpdate();
+        }
```
Comment (SMishAlex) [inline src/test/java/dao/CustomerDAOTest.java:30]: this is match better, you can also use your dao insert (and delete in after case) method to do it actually

**Review by SMishAlex — COMMENTED** [review]
good job

**Review by bvvvd — COMMENTED** [review]
it's ok
but refactor your too long lines, create some new variables to avoid hard-readable code

**Pull request merged**